src/Controller/Component/ObjectsEditorsComponent.php

```diff
@@ -0,0 +1,138 @@
+<?php
+namespace App\Controller\Component;
+
+use Cake\Cache\Cache;
+use Cake\Controller\Component;
+use Cake\Core\Configure;
+
+/**
+ * ObjectsEditors component
+ */
+class ObjectsEditorsComponent extends Component
+{
+    /**
+     * Objects editors.
+     *
+     * @var array
+     */
+    public $objectsEditors;
+
+    /**
+     * Concurrent check time.
+     *
+     * @var integer
+     */
+    public $concurrentCheckTime = 20000;
+
+    /**
+     * {@inheritDoc}
+     */
+    public function initialize(array $config): void
+    {
+        // set concurrent check time by config, if any. default 20s
+        $concurrentCheckTime = Configure::read('Editors.concurrentCheckTime');
+        if (!empty($concurrentCheckTime)) {
+            $this->concurrentCheckTime = $concurrentCheckTime;
+        }
+        // get concurrent editors
+        $this->objectsEditors = $this->getEditors();
+        // remove expired data
+        $this->cleanup();
+
+        parent::initialize($config);
+    }
+
+    /**
+     * Update objects editors adding the user access to ID object
+     *
+     * @param string $id The object ID
+     * @return string
+     */
+    public function update(string $id): string
+    {
+        if (!array_key_exists($id, $this->objectsEditors)) {
+            $this->objectsEditors[$id] = [];
+        }
+        $found = false;
+        $editor = $this->editorName();
+        $now = time();
+        foreach ($this->objectsEditors[$id] as &$objectEditor) {
+            if ($found) {
+                continue;
+            }
+            if ($objectEditor['name'] === $editor) {
+                $found = true;
+                $objectEditor['timestamp'] = $now;
+            }
+        }
+        if (!$found) {
+            $this->objectsEditors[$id][] = [
+                'name' => $editor,
+                'timestamp' => $now,
+            ];
+        }
```

Comment:
I suggest either:
```suggestion
        foreach ($this->objectsEditors[$id] as &$objectEditor) {
            if ($objectEditor['name'] === $editor) {
                $found = true;
                $objectEditor['timestamp'] = $now;
                break;
            }
        }
        if (!$found) {
            $this->objectsEditors[$id][] = [
                'name' => $editor,
                'timestamp' => $now,
            ];
        }
```

or

```suggestion
        $editorKey = array_search($editor, array_column($this->objectsEditors[$id], 'name'));
        if ($editorKey === false) {
            $this->objectEditors[$id][] = [
                'name' => $editor,
                'timestamp' => $now,
            ];
        } else {
            $this->objectEditors[$id][$editorKey]['timestamp'] = $now;
        }
```